### PR TITLE
refactor(payment_id): add payment id domain type

### PR DESCRIPTION
**Pull request by Narayanbhat166**
## Type of Change

- [x] Refactoring

## Description

This PR adds a domain type called `PaymentId` to be used in all of the codebase. 

## How did you test it?

- Compile the code 
- Postman tests on CI

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code

### CODE REVIEW

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 1]
```diff
@@ -314,8 +314,8 @@ impl TryFrom<&types::SetupMandateRouterData> for CreateCustomerProfileRequest {
                     create_customer_profile_request: AuthorizedotnetZeroMandateRequest {
                         merchant_authentication,
                         profile: Profile {
-                            //The payment ID is included in the description because the connector requires unique description when creating a mandate.
-                            description: item.payment_id.clone(),
+                            // The payment ID is included in the description because the connector requires unique description when creating a mandate.
+                            description: item.connector_request_reference_id.clone(),
```
Comment (sai-harsha-vardhan) [inline crates/router/src/connector/authorizedotnet/transformers.rs:318]: Did we handle this change else where, including webhooks?

File: crates/router/src/connector/checkout/transformers.rs [hunk 2]
```diff
@@ -1237,7 +1237,7 @@ pub struct CheckoutWebhookBody {
 #[derive(Debug, Serialize, Deserialize)]
 pub struct CheckoutDisputeWebhookData {
     pub id: String,
-    pub payment_id: Option<String>,
+    pub payment_id: Option<common_utils::id_type::PaymentId>,
```
Comment (sai-harsha-vardhan) [inline crates/router/src/connector/checkout/transformers.rs:1240]: This is not our payment_id, it should be string here

File: crates/router/src/connector/trustpay/transformers.rs [hunk 3]
```diff
@@ -1780,7 +1780,7 @@ impl TryFrom<WebhookStatus> for diesel_models::enums::RefundStatus {
 #[serde(rename_all = "PascalCase")]
 pub struct WebhookReferences {
     pub merchant_reference: String,
-    pub payment_id: Option<String>,
+    pub payment_id: Option<common_utils::id_type::PaymentId>,
```
Comment (sai-harsha-vardhan) [inline crates/router/src/connector/trustpay/transformers.rs:1783]: This is not our payment_id

File: crates/router/src/core/payments/helpers.rs [hunk 4]
```diff
@@ -2857,6 +2857,12 @@ pub async fn verify_payment_intent_time_and_client_secret(
         .async_map(|cs| async move {
             let payment_id = get_payment_id_from_client_secret(&cs)?;
 
+            let payment_id = id_type::PaymentId::try_from(Cow::Owned(payment_id)).change_context(
```
Comment (sai-harsha-vardhan) [inline crates/router/src/core/payments/helpers.rs:2860]: Can we have an impl function for converting string to PaymentId type?

**Review by jarnura — APPROVED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by ThisIsMani — APPROVED** [review]
Dashboard specific changes looks fine.

**Review by lsampras — APPROVED** [review]

**Review by ShankarSinghC — APPROVED** [review]

**Pull request merged**